tests/integration/support/steps.go

```diff
@@ -343,6 +365,67 @@ func (c *ScenarioCtx) PayloadsOfDispatchedEventsDontIncludeDecisions() error {
 	return nil
 }
 
+// TheUserProfileServiceStateShouldBe checks current state of UPS
+func (c *ScenarioCtx) TheUserProfileServiceStateShouldBe(value *gherkin.DocString) error {
+	config, err := c.clientWrapper.Client.GetProjectConfig()
+	if err != nil {
+		return fmt.Errorf("Invalid Project Config")
+	}
+	parsedProfiles, err := parseYamlArray(value.Content, config)
+	if err != nil {
+		return fmt.Errorf("Invalid request for user profile service state")
+	}
+
+	var expectedProfiles []decision.UserProfile
+	for _, profile := range parsedProfiles {
+		userProfile := decision.UserProfile{}
+		if userID, ok := profile["user_id"]; ok {
+			userProfile.ID = userID.(string)
+		}
+		if experimentBucketMap, ok := profile["experiment_bucket_map"]; ok {
+			userProfile.ExperimentBucketMap = make(map[decision.UserDecisionKey]string)
+			for k, v := range experimentBucketMap.(map[string]interface{}) {
+				decisionKey := decision.NewUserDecisionKey(k)
+				if bucketMap, ok := v.(map[string]interface{}); ok {
+					userProfile.ExperimentBucketMap[decisionKey] = bucketMap[decisionKey.Field].(string)
+				}
+			}
+		}
+		expectedProfiles = append(expectedProfiles, userProfile)
+	}
+
+	actualProfiles := c.clientWrapper.UserProfileService.(optlyplugins.UPSHelper).GetUserProfiles()
+
+	success := false
+	for _, expectedProfile := range expectedProfiles {
+		success = false
+		for index, actualProfile := range actualProfiles {
+			if subset.Check(expectedProfile, actualProfile) {
```

Comment:
Should this be a subset check? SHouldn't we check for strict equality between the actualProfiles and the expectedProfiles?

Reply:
This is because during conversion from map to array during parsing, the order of elements is not always accurate, because of which, direct subset comparison randomly fails. I have now implemented a check to validate the length of both arrays aswell.

Reply:
Currently, FSC also checks for subset comparison:
`Then("the User Profile Service state should be", function(expectedState) {
  // TODO: allow keys in this state
  expectedState = this.load(expectedState);
  expect(this.context.user_profiles).to.containSubset(expectedState);
});`
 i tried it with equality comparison but some of the tests because of greater number of profiles returned by the sdk.

Reply:
Isn't it a problem that we are returning a greater number of profiles than we expected with just the calls we have?

Reply:
@mikeng13 I think we do that in FSC to logically group the tests in Scenario Outlines. Each example test can result in a different user profile state at the end. So for equality check we will have to separate them out in multiple scenarios. 

I suggest that we modify the wording of this comparison. And write some tests based on equality.

Reply:
Responded on slack.